src/store/traction/pacbio/runCreate/mutations.js

```diff
@@ -1,11 +1,61 @@
 import { populateById } from '@/api/JsonApi'
+import { dataToObjectById } from '@/api/JsonApi'
+import Vue from 'vue'
+
+// Mutations handle synchronous update of state
+
+const mutations = {
+  setPools(state, pools) {
+    Vue.set(state, 'pools', {
+      ...state.pools,
+      ...dataToObjectById({ data: pools, includeRelationships: true }),
+    })
+  },
+  setPoolsForExistingRun(state, pools) {
+    const poolsObject = pools.reduce(function (acc, cur) {
+      acc[cur.id] = cur
+      return acc
+    }, {})
+    Vue.set(state, 'pools', {
+      ...state.pools,
+      ...poolsObject,
+    })
+  },
+  setTubes(state, tubes) {
```

Comment:
There are four functions all using the same structure. You could create a helper function which each one could then call.